GoogleForCreators/web-stories-wp
# Add border panel

## Summary



## Relevant Technical Choices



## To-do

- [x] ~~Displaying solid  border (currently dashed only)~~
- [x] Design in the panel
- [x] Output
- [x] Support for rectangle Shape element
- [ ] ~~Responsiveness in the FE~~
- [x] Proper support for media.
- [x] Adjustments once some UX details get confirmed (e.g. the lock behavior)



## User-facing changes
- There is a new Border panel which allows the user to add and customize border for elements.


## Testing Instructions
- Add an element (test with all element types)
- Add border width to it, verify that all sides will have the same width since the width is locked by default
- Unlock the border, change the widths, verify that it's changing as expected.
- Change color and opacity, verify it's assigned as expected.
- Change the border between inner-outer-center, verify it's displayed as expected. The inner border is inside the element and covers its content. The outer border is outside the element and doesn't cover its content. The center border is 50% outside, 50% inside.
- Verify that the border is correctly displayed in the front-end when saving the Story as well.
- Verify that an element that is not rectangular and has some other shape, does not display nor support border.

Note: The panel is using the previously existing design elements since redesign of the panels will happen later.


---



Fixes #1051
